src/Uno.Toolkit.UI/Behaviors/ItemsRepeaterExtensions.cs

```diff
@@ -474,18 +513,54 @@ private static bool SetSelectionStates(ItemsRepeater ir, params int[]? indexes)
 			SetSelectedItem(ir, items[0]);
 			SetSelectedIndexes(ir, indexes);
 			SetSelectedItems(ir, items);
+			SetSelectionInfoRanges(ir, indexes, diffIndexes);
 		}
 		else
 		{
 			SetSelectedIndex(ir, -1);
 			SetSelectedItem(ir, null);
 			SetSelectedIndexes(ir, Array.Empty<int>());
 			SetSelectedItems(ir, Array.Empty<object>());
+			SetSelectionInfoRanges(ir, Array.Empty<int>());
+
 		}
 
 		return true;
 	}
 
+	private static void SetSelectionInfoRanges(ItemsRepeater ir, int[] selectedIndexes, int[]? diffIndexes = null)
+	{
+		if (ir.ItemsSource is not ISelectionInfo selectionInfo) return;
+
+		IEnumerable<int>? indexesToDeselect = null;
+
+		// diffIndexes is used to optimize the selection change, by deselecting only the elements that are not in the new selection.
```

Comment:
```suggestion
		// diffIndexes is used to optimize the selection change, by updating only the elements that are not in the new selection.
```
not just select, but also deselect